### PR TITLE
FIX: allow content tags with mixed case to be added and the topic list to be correctly filtered

### LINKED ISSUE #27: tags are not getting added for some languages
There are some languages whose code includes capital letters (e.g zh_CN, en_US, etc). If a user selects it in the composer field and creates the topic; it doesn't add the corresponding tag to the topic.

hint: the 'underscore' method [here](https://github.com/paviliondev/discourse-multilingual/blob/main/lib/multilingual/language/content_tag.rb#L144
) turns it into downcase (e.g zh_cn, en_us, etc), and thus it cannot find the tag for it [here](https://github.com/paviliondev/discourse-multilingual/blob/main/lib/multilingual/language/content_tag.rb#L146)

### PR DESCRIPTION
**Pull request by merefield**

### CODE REVIEW

**Comment by angusmcleod** [comment]
@merefield I'm just wondering why this was underscored in the first place. Not saying we shouldn't support uppercase, it's needed, but what was the reason for underscoring? Also will this be backwards compatible with existing content tags?

**Comment by merefield** [comment]
@angusmcleod I do not see any such string manipulation in core wrt to tags: https://github.com/discourse/discourse/search?l=Ruby&p=1&q=underscore

In addition it's not impacting any tests?

It is an unusual method to use on a string, which suggests there was some purpose to it, but the reason eludes me?  Not being the author doesn't help either.  I say we give it a go ... (PS changed the title of the PR, it was late! 😅 )

**Comment by angusmcleod** [comment]
I'm willing to see how we go with it functionality wise, but are we sure it's backwards compatible?

**Comment by merefield** [comment]
@angusmcleod as discussed on chat, checked and discovered topic list filtering is also broken wrt capitalisation in content tags, so have fixed that and modified associated test to ensure that scenario is captured.

please note I'm technically filtering on content tags with underscores to find the first test candidate, but these all contain capital letters as that appears to be the convention.

File: spec/components/topic_query_spec.rb [hunk 1]
```diff
@@ -39,11 +39,11 @@
     }
 
     before do
-      user1.custom_fields['content_languages'] = ['aa', 'ab']
+      user1.custom_fields['content_languages'] = [Multilingual::ContentTag.all.first, Multilingual::ContentTag.all.select { |t| t.include?("_") }.first]
       user1.save_custom_fields(true)
     end
 
-    it "filters topic list when content language topic filtering is enabled" do
+    it "filters topic list when content language topic filtering is enabled, whilst supporting capitalisation in content tags" do
```
Comment (angusmcleod) [inline spec/components/topic_query_spec.rb]: @merefield Sorry to be persnickety but checking it works with capitalisation should really be a separate test.

**Review by angusmcleod — CHANGES_REQUESTED** [review]

File: spec/components/topic_query_spec.rb (hunk 1 above)
Comment (angusmcleod) [reply spec/components/topic_query_spec.rb]: It will make the test case clearer.

Comment (merefield) [reply spec/components/topic_query_spec.rb]: I've added an additional test case

File: spec/components/topic_query_spec.rb [hunk 2]
```diff
@@ -39,22 +45,32 @@
...
       SiteSetting.multilingual_content_languages_topic_filtering_enabled = false
 
-      expect(TopicQuery.new(user1).list_latest.topics.count).to eq(4)
-      expect(TopicQuery.new(user2).list_latest.topics.count).to eq(4)
+      expect(TopicQuery.new(user1).list_latest.topics.count).to eq(5)
+      expect(TopicQuery.new(user2).list_latest.topics.count).to eq(5)
+      expect(TopicQuery.new(user2).list_latest.topics.count).to eq(5)
     end
+
+    it "filters topic list when content language topic filtering is enabled and includes content tags with capitalisation" do
```
Comment (angusmcleod) [inline spec/components/topic_query_spec.rb:69]: @merefield the changes look good, but this is still testing two things?

**Review by angusmcleod — APPROVED** [review]

**Pull request merged**